embedchain/eval/metrics/context_relevancy.py

```diff
@@ -0,0 +1,88 @@
+import os
+from string import Template
+from typing import Optional
+
+import numpy as np
+import pysbd
+from openai import OpenAI
+
+from ..base import BaseMetric
+from ..config import EvalConfig
+from ..utils import EvalData
+
+CONTEXT_RELEVANCY_PROMPT = Template(
+    """
+                            Please extract relevant sentences from the provided context that is required to answer the given question.
+                            If no relevant sentences are found, or if you believe the question cannot be answered from the given context, return the empty string ("").
+                            While extracting candidate sentences you're not allowed to make any changes to sentences from given context or make up any sentences.
+                            You must only provide sentences from the given context and nothing else.
+                            
+                            Context: $context
+                            Question: $question
+                            """  # noqa: E501
+)
+
+
+class ContextRelevance(BaseMetric):
+    """
+    Metric for context relevance.
+
+    This evaluator evaluates the context relevance of a model.
+    """
+
+    def __init__(self, config: Optional[EvalConfig] = None):
+        super().__init__()
+        self.config = config or EvalConfig()
+        api_key = self.config.api_key or os.environ["OPENAI_API_KEY"]
+        if not api_key:
+            raise ValueError("Please set the OPENAI_API_KEY environment variable or pass the `api_key` in config.")
+        self.client = OpenAI(api_key=api_key)
+        self.sbd = pysbd.Segmenter(language=self.config.language, clean=False)
```

Comment:
Not sure if this `self.sbd` really needs to be an attribute of the metric. It seems like an internal thing which user won't even have to care about / access it.